src/metabase/api/pulse.clj

```diff
@@ -117,6 +116,35 @@
   ;; return updated Pulse
   (pulse/retrieve-pulse id))
 
+(api/defendpoint GET "/form_input_refresh"
+  "Provides relevant configuration information and user choices for creating/updating Pulses, can block for quite a long
+  amount of time if the conversations-list or the users-list is huge."
+  []
+  (let [chan-types (-> channel-types
```

Comment:
It seems like a malicious user could hit this endpoint a bunch of times over and over and nuke an instance -- it seems like we should probably limit this somehow. If someone hit refresh 10 seconds ago there is nothing another refresh is likely to do anyway, right?